src/Hamiltonians/HubbardMomSpace.jl

```diff
@@ -0,0 +1,639 @@
+"""
+    dispersion_MomSpace(ks, geometry, t)
+    Dispersion relation for a given set of k values, lattice geometry and hopping strengths t. Returns
+        ``-2(\\sum_{\\bar{k}} \\Re(t_{\\bar{k}}) \\cos(k_{\\bar{k}}) + \\Im(t_{\\bar{k}}) \\sin(k_{\\bar{k}}))``
+    where ``\\bar{k}`` runs over all dimensions of the lattice.
+"""
+function dispersion_MomSpace(ks::SVector{D}, geometry::CubicGrid{D, S}, t::SMatrix) where {D,S}
+    # Calculate the dispersion relation for a given set of k values and hopping strength t.
+    C,_ = size(t)
+    M = prod(S)
+    kes_mat = zeros(Float64,C,M)
+    ks_mat = zeros(Float64,D,M)
+    phase = atan.(imag.(t)./real.(t))
+    for j in 1:C
+        for i in 1:M
+            mom_val = value_of_mom_mode(M-i+1, ks, geometry)
+            kes_mat[j,i] = convert(Float64,hub_dis_Mom_Space(t[j,:], mom_val)[1])
+            ks_mat[:,i] = mom_val + phase[j,:]
+        end
+    end
+    return SMatrix{C,M,Float64}(kes_mat), SMatrix{D,M,Float64}(ks_mat)
+end
+function value_of_mom_mode(add_index::Int, ks::SVector{D}, geometry::CubicGrid{D}) where {D}
+    mom_mode = reverse(geometry[add_index])
+    return [ks[i][mode] for (i, mode) in enumerate(mom_mode)]
+end
+
+function hub_dis_Mom_Space(t::SVector, k::Vector)
```

Comment:
```suggestion
function hub_dis_mom_space(t::SVector, k::Vector)
```
Don't use capitals in function names.